pkg/segment/query/processor/searcher.go

```diff
@@ -0,0 +1,366 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package processor
+
+import (
+	"math"
+	"sort"
+	"time"
+
+	"github.com/siglens/siglens/pkg/segment/query"
+	_ "github.com/siglens/siglens/pkg/segment/query"
+	"github.com/siglens/siglens/pkg/segment/query/iqr"
+	"github.com/siglens/siglens/pkg/segment/query/summary"
+	"github.com/siglens/siglens/pkg/segment/results/segresults"
+	"github.com/siglens/siglens/pkg/segment/structs"
+	segutils "github.com/siglens/siglens/pkg/segment/utils"
+	toputils "github.com/siglens/siglens/pkg/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+type block struct {
+	*structs.BlockSummary
+	*structs.BlockMetadataHolder
+	parentSSR *structs.SegmentSearchRequest
+	filename  string
```

Comment:
nit: what is this filename for ? is it a segkey filename or a csg filename or something else ?

Reply:
I think it's the segkey filename, but I'm not 100% sure. It's needed to build the `map[string]SegmentSearchRequest` parameter to `ApplyFilterOperatorInternal()`. The new code in this PR doesn't directly use the filename; it's computed from `GetSSRsFromQSR()`, then saved to each of the blocks I make, and then read from the blocks to rebuild the `map[string]SegmentSearchRequest`.

Do you think I should rename it?